jaeger-core/src/main/java/com/uber/jaeger/metrics/Metrics.java

```diff
@@ -80,10 +80,12 @@ public static String addTagsToMetricName(String name, Map<String, String> tags)
 
     SortedSet<String> sortedKeys = new TreeSet<String>(tags.keySet());
     for (String key : sortedKeys) {
-      sb.append(".");
-      sb.append(key);
-      sb.append("=");
-      sb.append(tags.get(key));
+      if (!tags.get(key).isEmpty()) {
```

Comment:
nit: there are two accesses to the map. `SortedMap` can be used Instead of `SortedSet` and then just iterate over entries.

Reply:
I don't think this belongs to this PR. Could you open an issue to change this?

Reply:
I will directly open a PR. 

Why does it have to be sorted?